generate_tidal_profile.py

```diff
@@ -0,0 +1,438 @@
+        ax1.set_ylabel('Power (kW)')
+        ax2 = fig.add_subplot(122)
+        self.power_profiles[min_profile_num].plot(
+            ax=ax2, title='Profile with min energy generation')
+        ax2.set_ylabel('Power (kW)')
+
+    def get_dc_to_ac(self):
+        """ Returns the dc to ac ratio. """
+
+        # Should we keep this? I don't think this matters
+        # for tidal as much for solar
+
+        # Get DC power output
+        dc = self.advanced_inputs['rated_power']
+
+        # Get inverter AC power output
+        ac = self.advanced_inputs['inverter_output']
+
+        return dc / ac
+
+    def get_turbine_losses(self):
+        """ Returns the system losses. """
+
+        # I don't think this is necessary either, since losses are
+        # an input, not an output like for solar
+
+        turbine_losses = self.advanced_inputs['turbine_losses']
+
+        return turbine_losses
+
+
+def calc_tidal_prod(tidal_profile, latitude, longitude,
+                    rated_power, rotor_length, turbine_number,
+                    inverter_efficiency, maximum_cp, turbine_losses,
+                    cut_in_velocity, cut_out_velocity, validate=False):
+
+    """ Calculates the production from a tidal profile. """
+
+    if validate:
+        # Put arguments in a dict
+        args_dict = {'tidal_profile': tidal_profile,
+                     'latitude': latitude,
+                     'longitude': longitude,
+                     'rated_power': rated_power,
+                     'rotor_length': rotor_length,
+                     'turbine_number': turbine_number,
+                     'inverter_efficiency': inverter_efficiency,
+                     'maximum_cp': maximum_cp,
+                     'turbine_losses': turbine_losses,
+                     'cut_in_velocity': 0.7,
+                     'cut_out_velocity': 2.6}
+    #
+        # Validate all parameters
+        validate_all_parameters(args_dict)
+
+
+    # Calculate DC power (normalized to turbine size. i.e. per 1kW of tidal)
+
+    dc_power = pd.DataFrame()
+
+    for index, row in tidal_profile.iterrows():
+        u = row['v']
+        if u > cut_in_velocity and u < cut_out_velocity:
+            dc_power.at[
+                index, 'power'] = 0.5 * maximum_cp * u ** 3 * np.pi * rotor_length ** 2 / (rated_power * turbine_number)
+        elif u < cut_in_velocity:
+            dc_power.at[index, 'power'] = 0
+        elif u > cut_out_velocity:
+            dc_power.at[index, 'power'] = rated_power / (rated_power * turbine_number)
+        else:
+            dc_power.at[index, 'power'] = np.nan
+
+    # Calculate losses
+    dc_power['power'] = dc_power['power'] * (1 - turbine_losses / 100)
+
+
+    # Calculate AC power
+    ac_power = dc_power['power'] * inverter_efficiency
+
+    # Force values less than 0 to 0
+    ac_power[ac_power < 0] = 0
+
+    # Divide by the total DC power to get the power (in kW) per 1kW of tidal
+    return ac_power / rated_power
```

Comment:
Should this also be divided by # turbines?

Reply:
I actually don't think this line is required anymore, since the dc power calculation already normalizes to 1kW of tidal